src/test/controllers/localStorage_test.ts

```diff
@@ -103,4 +103,43 @@ suite('LocalStorageController', () => {
       assert.is(storageValue, '[5,6]');
     });
   });
+
+  suite('with saved value', () => {
+    setup(() => {
+      window.localStorage.setItem('test-key', JSON.stringify([7, 8]));
+      element = document.createElement('test-element') as TestElement;
+      element.template = () => html`${JSON.stringify(controller.value)}`;
+      document.body.appendChild(element);
+    });
+
+    suite('without default value', () => {
+      setup(async () => {
+        controller = new LocalStorageController(element, 'test-key');
+        element.controllers.push(controller);
+        await element.updateComplete;
+      });
+
+      test('initialises to saved value', () => {
+        const storageValue = window.localStorage.getItem('test-key');
+        assert.equal(controller.value, [7, 8]);
+        assert.equal(element.shadowRoot!.textContent, '[7,8]');
+        assert.is(storageValue, '[7,8]');
+      });
+
+      suite('with default value', () => {
```

Comment:
i think you meant to close the previous suite off before this one (so both are a child of `with saved value`)

Reply:
No, I use the suite 'with saved value' as umbrella for with and without default value

Reply:
Maybe something like this is better:

```
suite('with saved value', () => {
      setup(() => {
        window.sessionStorage.setItem('test-key', JSON.stringify([7, 8]));
        element = document.createElement('test-element') as TestElement;
        element.template = () => html`${JSON.stringify(controller.value)}`;
        document.body.appendChild(element);
      });

      test('initialises to saved value', async () => {
        controller = new SessionStorageController(element, 'test-key');
        element.controllers.push(controller);
        await element.updateComplete;

        const storageValue = window.sessionStorage.getItem('test-key');
        assert.equal(controller.value, [7, 8]);
        assert.equal(element.shadowRoot!.textContent, '[7,8]');
        assert.is(storageValue, '[7,8]');
      });

      test('ignore default value and use saved value', async () => {
        controller = new SessionStorageController(element, 'test-key', [5, 6]);
        element.controllers.push(controller);
        await element.updateComplete;

        const storageValue = window.sessionStorage.getItem('test-key');
        assert.equal(controller.value, [7, 8]);
        assert.equal(element.shadowRoot!.textContent, '[7,8]');
        assert.is(storageValue, '[7,8]');
      });
    });
    ```

Reply:
i think we're in agreement! the code you had was this structure:

- with saved value
  - sub-suite A
    - sub-suite B

i just meant it probably was meant to be:

- with saved value
  - sub-suite A
  - sub-suite B

(i.e. using `with saved value` as an umbrella like you said)

what you pushed also works though! either is fine